crescent/commands/errors.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable, Generic, Protocol, Type, TypeVar, overload
+
+from crescent.internal.meta_struct import MetaStruct
+
+if TYPE_CHECKING:
+    from crescent.context import Context
+    from crescent.internal.app_command import AppCommandMeta, CommandCallback
+    from crescent.typedefs import CommandOptionsT
+
+
+ERROR = TypeVar("ERROR", bound=Exception, contravariant=True)
+
+
+class ErrorHandlerProto(Protocol, Generic[ERROR]):
+    async def __call__(
+        self,
+        /,
+        *,
+        exc: ERROR,
+        ctx: Context,
+        command: MetaStruct[CommandCallback, AppCommandMeta],
```

Comment:
`MetaStruct` should only be used internally. Its too sketchy to give that to users. Either the callback should be passed in or only give `exc`, `ctx`, and `options`.

Reply:
I see your point, but I'm not sure what else to do. Just passing the callback means that they can't access stuff like the real command name (if passed).

Reply:
They shouldn't need to access anything besides group, sub group, and all which should all be on ctx anyway.

Reply:
What about command name?

Reply:
Yeah, that one should be in ctx too.